code/components/jomjol_flowcontroll/MainFlowControl.cpp

```diff
@@ -474,6 +474,81 @@ esp_err_t handler_json(httpd_req_t *req)
+ * 
+ * A MetricFamily with a Metric for each Sequence is provided. If no valid value is available, the metric is not provided.
+ * MetricPoints are provided without a timestamp. Additional metrics with some device information is also provided.
+ * 
+ * The default metric prefix is 'ai_on_the_edge_device_'.
+ * 
+ * example configuration for Prometheus (`prometheus.yml`):
+ * 
+ *    - job_name: watermeter
+ *      static_configs:
+ *        - targets: ['watermeter.fritz.box']
+ * 
+*/
+esp_err_t handler_openmetrics(httpd_req_t *req)
+{
+#ifdef DEBUG_DETAIL_ON
+    LogFile.WriteHeapInfo("handler_openmetrics - Start");
+#endif
+
+    ESP_LOGD(TAG, "handler_openmetrics uri: %s", req->uri);
+
+    if (bTaskAutoFlowCreated)
+    {
+        httpd_resp_set_hdr(req, "Access-Control-Allow-Origin", "*");
+        httpd_resp_set_type(req, "text/plain"); // application/openmetrics-text is not yet supported by prometheus
+
+        const string metricNamePrefix = "ai_on_the_edge_device_";   // FIXME: move to config-file
+
+        // get current measurement (flow)
+        string response = flowctrl.getOpenMetrics(metricNamePrefix);
+
+        // CPU Temperature
+        response += "# HELP " + metricNamePrefix + "cpu_temperature_celsius current cpu temperature in celsius\n# TYPE " 
+            + metricNamePrefix + "cpu_temperature_celsius gauge\n" 
+            + metricNamePrefix + "cpu_temperature_celsius " + std::to_string((int)temperatureRead()) + "\n";
+
+        // WiFi signal strength
+        response += "# HELP " + metricNamePrefix + "rssi_dbm current WiFi signal strength in dBm\n# TYPE " 
+            + metricNamePrefix + "rssi_dbm gauge\n" 
+            + metricNamePrefix + "rssi_dbm " + std::to_string(get_WIFI_RSSI()) + "\n";
+
+        // memory info
+        response += "# HELP " + metricNamePrefix + "memory_heap_free_bytes available heap memory\n# TYPE " 
+            + metricNamePrefix + "memory_heap_free_bytes gauge\n" 
+            + metricNamePrefix + "memory_heap_free_bytes " + std::to_string(getESPHeapSize()) + "\n";
+
+        // device uptime
+        response += "# HELP " + metricNamePrefix + "uptime_seconds device uptime in seconds\n# TYPE " 
+            + metricNamePrefix + "uptime_seconds gauge\n" 
+            + metricNamePrefix + "uptime_seconds " + std::to_string((long)getUpTime()) + "\n";
+
+        // data aquisition round
+        response += "# HELP " + metricNamePrefix + "rounds_total data aquisition rounds since device startup\n# TYPE " 
+            + metricNamePrefix + "rounds_total counter\n" 
+            + metricNamePrefix + "rounds_total " + std::to_string(countRounds) + "\n";
+
+        // the response always contains at least the metadata (HELP, TYPE) for the MetricFamily so no length check is needed
+        httpd_resp_send(req, response.c_str(), response.length());
+    }
+    else
+    {
+        httpd_resp_send_err(req, HTTPD_403_FORBIDDEN, "Flow not (yet) started: REST API /metrics not yet available!");
```

Comment:
HTTP 4xx is a client-side error. HTTP 5xx, in particular [HTTP 503](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/503) seems more appropriate for this case.

Reply:
While I totally agree with you, the [enum with the response codes](https://github.com/espressif/esp-idf/blob/v5.2/components/esp_http_server/include/esp_http_server.h#L555) does not provide a 503. I guess that's the reason why all the other handler also use 403 instead.

Reply:
I'd say `HTTPD_500_INTERNAL_SERVER_ERROR` would've been a better choice given the available codes (it's still indicative of a server problem), but if all other handlers use 403 as well your code is at least consistent with those.